src/include/sof/lib/uuid.h

```diff
@@ -0,0 +1,46 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2020 Intel Corporation. All rights reserved.
+ *
+ * Author: Marcin Maka <marcin.maka@linux.intel.com>
+ */
+
+#ifndef __SOF_LIB_UUID_H__
+#define __SOF_LIB_UUID_H__
+
+#include <stdint.h>
+
+#define UUID_SIZE 16
+
+struct sof_uuid {
+	uint32_t a;
+	uint16_t b;
+	uint16_t c;
+	uint8_t  d[8];
+};
```

Comment:
Maybe a small explanation in the comments about how this is formed and constructed. Doxygen should be able to pick this up too.

Reply:
Brief description added.